module/dal/remote/src/main/java/org/openbase/bco/dal/remote/layer/unit/Units.java

```diff
@@ -384,9 +384,9 @@ private static void resetUnitRegistryObserver() throws CouldNotPerformException
      * @throws InterruptedException  is thrown if the current thread was
      *                               externally interrupted.
      */
-    private static UnitRemote<?> getUnitRemote(final String unitId) throws NotAvailableException, InterruptedException {
+    private static UnitRemote<? extends Message> getUnitRemote(final String unitId) throws NotAvailableException, InterruptedException {
```

Comment:
```suggestion
    private static UnitRemote<D extends Message> getUnitRemote(final String unitId) throws NotAvailableException, InterruptedException {
```

Why did you not change this here?

Reply:
The Datatype is not fix (it can be a remote of any unit type and the only thing we know is that the type is a protobuf message), thus even though I would introduce a generic variable `D` java would again elimite it during compile time. However, that would lead to an additional (not necessary) type cast that lowers the performance.
In Java its typical to use these `? extends XXX` declaration to refer a datatype in case only the super cass is known.